sklearn/pipeline.py

```diff
@@ -239,6 +244,15 @@ def _final_estimator(self):
         estimator = self.steps[-1][1]
         return 'passthrough' if estimator is None else estimator
 
+    def _log_message(self, step_idx):
+        if not self.verbose:
+            return
+        n_steps = len([est for _, est in self.steps if est is not None])
+        step_num = len(
+            [est for _, est in self.steps[:step_idx + 1] if est is not None])
+        return '(step %d of %d) Fitting %s' % (step_num, n_steps,
+                                               self.steps[step_idx][0])
```

Comment:
Could be wrong but I'm not sure this is correct: you're indexing `self.steps` with `step_idx`, but in `_fit` `step_idx` it actually an idx from `_iter()` where the None or passthrough transformer have already been filtered out. So this `self.steps[step_idx][0]` name could refer to the wrong one?


Slightly related: no strong opinion but this numbering scheme is ignoring the None/passthrough steps, which is a bit inconsistent with the `named_steps` attribute.

Reply:
`_iter` keeps track of the index with respect to `self.steps` correctly. So `step_idx` in this context would refer to the correct step.

This PR was updated with a custom log message for passthrough/drop:

```
[Pipeline] ............... (step 1 of 3) Fitting transf, total=   0.0s
[Pipeline] ................. (step 2 of 3) Passing noop, total=   0.0s
[Pipeline] .................. (step 3 of 3) Fitting clf, total=   0.0s
```

Reply:
oh right, thanks